src/redux/reducers/samples/experimentsMetadataRename.js

```diff
@@ -0,0 +1,25 @@
+/* eslint-disable no-param-reassign */
+import produce, { original } from 'immer';
+
+const experimentsMetadataRename = produce((draft, action) => {
+  const { oldKey, newKey, experimentId } = action.payload;
+
+  const { meta, ...samples } = original(draft);
+
+  const sampleEntries = Object.entries(samples);
+
+  const sampleIdsFromExperiment = sampleEntries
+    .filter(([, { experimentId: currentExpId }]) => currentExpId === experimentId)
+    .map(([sampleId]) => sampleId);
+
+  // Move the metadata in oldKey to newKey and then remove the old one
+  sampleIdsFromExperiment.forEach((sampleId) => {
+    const value = draft[sampleId].metadata[oldKey];
+
+    delete draft[sampleId].metadata[oldKey];
+
+    draft[sampleId].metadata[newKey] = value;
```

Comment:
The fix for [the bug](https://github.com/hms-dbmi-cellenics/ui/pull/776/files#r925001865) is moved here: 

We store the value in a temporary variable, delete the `oldKey` and then set the new one, this ensures it will work even if `oldKey === newKey`

Reply:
why not just if "oldKey" == "newKey", do not update?

Reply:
it seemed safer to implement a simple solution that covers all cases rather than have a guard for the case where it breaks and then have the rest of the solution